osu.Game/IO/CopyOnWriteHelper.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace osu.Game.IO
+{
+    internal static class CopyOnWriteHelper
+    {
+        public static bool CheckAvailability(string testDestinationPath, string testSourcePath)
+        {
+            if (OperatingSystem.IsWindowsVersionAtLeast(10, 0, 26100))
+            {
+                // Starting from Windows 11 24H2, the CopyFile syscall will perform CoW cloning on ReFS drives.
+
+                const string test_filename = "_cow_test";
```

Comment:
Is it possible to open the directories themselves, to avoid creating random files?

Reply:
It seems actually possible. https://github.com/microsoft/CopyOnWrite uses a complex solution that enumerates all the volumes on the system, then find the volume the path belongs to. There is a syscall `GetVolumePathName` to map a directory to the volume, and `GetVolumeInformation` to query the volume information by its root path. I'll test it tomorrow.

Reply:
No need for the complex solution, just open `D:\osu\Songs` and not `D\osu\Songs\_cow_test`.

Reply:
Learned a bit about directory handle. It can be opened with specialized `CreateFile` and passed to `GetVolumeInformationByHandle`. Sounds actionable.